shared/connection.go

```diff
@@ -61,8 +62,10 @@ func MongoSession(opts MongoSessionOpts) *mgo.Session {
 		return nil
 	}
 
-	dialInfo.Direct = true // Force direct connection
+	// connect directly, fail faster, do not retry - for faster responses and accurate metrics, including mongoUp
+	dialInfo.Direct = true
 	dialInfo.Timeout = dialMongodbTimeout
+	dialInfo.FailFast = true
```

Comment:
Makes sense, but why we then in other parts of the code retry on failed queries `for tries < 2 {`? (yeah, I know it's not a part of this PR, just a note)

Reply:
Good point. We should remove them.